sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3681,6 +3681,48 @@ object GpuOverrides extends Logging {
         TypeSig.ARRAY + TypeSig.DECIMAL_64).nested(), TypeSig.all),
       (sample, conf, p, r) => new GpuSampleExecMeta(sample, conf, p, r)
     ),
+    exec[SubqueryBroadcastExec](
+      "Plan to collect and transform the broadcast key values",
+      ExecChecks(TypeSig.all, TypeSig.all),
+      (s, conf, p, r) => new SparkPlanMeta[SubqueryBroadcastExec](s, conf, p, r) {
+
+        override val childExprs: Seq[BaseExprMeta[_]] = Nil
+
+        override val childPlans: Seq[SparkPlanMeta[SparkPlan]] = Nil
+
+        override def tagPlanForGpu(): Unit = s.child match {
+          case ex @ BroadcastExchangeExec(_, c2r: GpuColumnarToRowExecParent) =>
+            val exMeta = new GpuBroadcastMeta(ex.copy(child = c2r.child), conf, p, r)
+            exMeta.tagForGpu()
+            if (!exMeta.canThisBeReplaced) {
+              willNotWorkOnGpu("underlying BroadcastExchange can not run on the GPU.")
+            }
+          case _ =>
+            willNotWorkOnGpu("no available BroadcastExchange for reuse.")
+        }
+
+        // Simply returns the original plan. Because its only child, BroadcastExchange, doesn't
+        // need to change if SubqueryBroadcastExec falls back to the CPU.
+        override def convertToCpu(): SparkPlan = s
+
+        // The rule PlanDynamicPruningFilters will insert SubqueryBroadcastExec if there exists
+        // available broadcast exchange for reuse. The plan stack of SubqueryBroadcastExec:
+        //      SubqueryBroadcast -> BroadcastExchange -> executedPlan
+        // Since the GPU overrides rule has been applied on executedPlan, the plan stack become:
+        //      SubqueryBroadcast -> BroadcastExchange -> GpuColumnarToRow -> GpuPlanStack...
+        // , if the wrapped subquery can run on the GPU.
+        // To reuse BroadcastExchange on the GPU, we shall transform above pattern into:
+        //      GpuSubqueryBroadcast -> GpuBroadcastExchange -> GpuPlanStack...
+        override def convertToGpu(): GpuExec = s.child match {
+          case ex @ BroadcastExchangeExec(_, c2r: GpuColumnarToRowExecParent) =>
+            val exMeta = new GpuBroadcastMeta(ex.copy(child = c2r.child), conf, p, r)
```

Comment:
nit: we do this twice. Once to tag and once here. It would be nice if we could cache it so we are not wasting work in the common case.

Reply:
Refined.